omeroweb/webgateway/static/3rdparty/jquery.jstree-3.0.8/jstree.js

```
@@ -7066,7 +7066,7 @@
 		};
 		// proto.attributeChangedCallback = function (name, previous, value) { };
 		try {
-			document.registerElement("vakata-jstree", { prototype: proto });
+			window.customElements.define("vakata-jstree", { prototype: proto });
```

Are there any other changes to jquery.jstree that we should consider? i.e. does this need bumping?

---

I don't think it *needs* bumping or that we are currently looking to add features that require newer releases. So not high priority just now, but could look into it if anything changes...

---

https://www.chromestatus.com/features/4642138092470272 which was removed in M80. 

Reference post:

 * https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade

M80 was released on 4 Feb so currently this functionality likely silently fails or throws hard JavaScript errors. There is at least one other location in jstree where `registerElement` is called and it is also called within panojs. We should most definitely look at a jstree, and possible a panojs, upgrade and/or full patching.

/cc @knabar, @jburel

---

jstree issue for further reference:

 * https://github.com/vakata/jstree/issues/2094

---

line 7084 is also more likely to create issue
`` if(document.registerElement ``

---

@jburel That was also pointed out on the issue that Chris linked: https://github.com/vakata/jstree/issues/2094#issuecomment-433926602
But the issue hasn't been re-opened. I tested, removing that whole block of code and it doesn't break any jsTree functionality. It creates a ```createdCallback()``` function but I can't see that function called anywhere or find it in the docs.

---

In that case, maybe put a note so we know we can remove it later on

---

Done